### PR TITLE
Make PS compatible with PHP 7.2 (1.7.3.x)

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.3.x
| Description?  | When testing PrestaShop with PHP 7.2, many errors appeared in the PHP error log. It seems the community already started to use that version with PrestaShop, so we try to set the base as early as possible.
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | Nope
| Deprecations? | Nope
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-4444
| How to test?  | Test on a server running with PHP 7.2.

Related PRs:
- [x] https://github.com/PrestaShop/ps_facetedsearch/pull/30

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8799)

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Bundle of:
* #8782
* #8783 
* #8784

PRs to include:
* #8584

**Comment by nalysius** [comment]
Also linked with #8760 I think.

**Comment by Quetzacoalt91** [comment]
This one is for another branch, that's why I did not mention it. :)

**Comment by nalysius** [comment]
I hadn't seen that, sorry :)

File: composer.lock [hunk 1]
```diff
@@ -8,16 +8,16 @@
     "packages": [
         {
             "name": "beberlei/DoctrineExtensions",
-            "version": "v1.0.19",
```
Comment (eternoendless) [inline composer.lock:11]: Any particular reason to update dependencies?

Comment (Quetzacoalt91) [reply composer.lock:11]: Yep, I wanted to get the modules compatible with PHP 7.2. Upgraded with the merge of https://github.com/PrestaShop/ps_facetedsearch/pull/30

File: src/PrestaShopBundle/Api/QueryParamsCollection.php [hunk 2]
```diff
@@ -657,6 +657,10 @@ protected function appendSqlSearchFilter($filters)
             return $filters;
         }
 
+        if (!is_array($this->queryParams['filter']['keywords'])) {
+            $this->queryParams['filter']['keywords'] = (array)$this->queryParams['filter']['keywords'];
```
Comment (eternoendless) [inline src/PrestaShopBundle/Api/QueryParamsCollection.php:661]: It's generally best to handle this case like this:
```php
$this->queryParams['filter']['keywords'] = array($this->queryParams['filter']['keywords']);
```
Casting an object into array will transform it into a hash-map: https://3v4l.org/ihISI

File: travis-scripts/base64-screenshots [hunk 3]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+FILES=$TRAVIS_BUILD_DIR/tests/Selenium/errorShots/*
+for f in $FILES
+do
+  echo "--- Displaying content of $f ...\n"
+
+  echo "data:image/png;base64,$(base64 -w 0 $f)"
+done
```
Comment (eternoendless) [inline travis-scripts/base64-screenshots:9]: Unrelated to PHP 7.2 but nice extra 😁

File: src/PrestaShopBundle/Api/QueryParamsCollection.php (hunk 2 above)
Comment (Quetzacoalt91) [reply src/PrestaShopBundle/Api/QueryParamsCollection.php:661]: Yeah this is what I wanted to do. Do you need a dump of the current value?

File: composer.lock (hunk 1 above)
Comment (eternoendless) [reply composer.lock:11]: In the future you can update PrestaShop's own dependencies only by running
```
composer update "prestashop/*"
```
This way we won't risk regressions from updated dependencies which are unrelated

File: src/PrestaShopBundle/Api/QueryParamsCollection.php (hunk 2 above)
Comment (eternoendless) [reply src/PrestaShopBundle/Api/QueryParamsCollection.php:661]: If it's a hash-map you wanted, it's fine by me 👍

**Review by eternoendless — DISMISSED** [review]

**Comment by marionf** [comment]
@Quetzacoalt91 

I have an issue at the first step of the installation

capture du 2018-03-07 16-44-37

File: travis-scripts/base64-screenshots (hunk 3 above)
Comment (mickaelandrieu) [reply travis-scripts/base64-screenshots:9]: agree: love it!

**Comment by Quetzacoalt91** [comment]
Can't fix it with the current version of symfony/monolog-bundle.
Closing.

**Comment by eternoendless** [comment]
@Quetzacoalt91 Can we keep the base 64 image for travis?

**Comment by Quetzacoalt91** [comment]
Sure, I can provide a specific PR on the branch 1.7.3.x for that feature.

**Pull request closed without merging**